packages/realm/bindgen/src/templates/react-native-wrapper.ts

```diff
@@ -27,22 +27,33 @@ export function generate(context: TemplateContext): void {
 
   out(`
     /*global global*/
-    import { Platform, NativeModules } from "react-native";
-    if (Platform.OS === "android") {
-      // Getting the native module on Android will inject the Realm global
-      // eslint-disable-next-line @typescript-eslint/no-unused-vars
-      const RealmNativeModule = NativeModules.Realm;
+    import { Platform, TurboModuleRegistry, NativeModules } from "react-native";
+
+    function getBinding() {
+      const NativeRealmModule = TurboModuleRegistry.get("Realm");
+      if (NativeRealmModule) {
+        return NativeRealmModule.getBinding();
+      }
+      if (Platform.OS === "android") {
+        // Getting the native module on Android will inject the Realm global
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        const RealmNativeModule = NativeModules.Realm;
+      }
+      const { __injectedRealmBinding } = global;
+      if (__injectedRealmBinding) {
+        return __injectedRealmBinding;
+      }
+      throw new Error("Could not find the Realm native module. Please consult our troubleshooting guide: https://www.mongodb.com/docs/realm-sdks/js/latest/#md:troubleshooting-missing-binary");
     }
-    // TODO: Remove the need to store Realm as a global
-    // @see https://github.com/realm/realm-js/issues/2126
-    const nativeModule = global.__RealmFuncs;
-    if(!nativeModule) {
-      throw new Error("Could not find the Realm binary. Please consult our troubleshooting guide: https://www.mongodb.com/docs/realm-sdks/js/latest/#md:troubleshooting-missing-binary");
+
+    const nativeBinding = getBinding();
```

Comment:
I've renamed this from `nativeModule`, since the binding is returned from a function call on the "native module" and is not the module itself.